include/boost/url/detail/remove_dot_segments.hpp

```diff
@@ -28,8 +28,8 @@ int
 normalized_path_compare(
     string_view lhs,
     string_view rhs,
-    bool remove_unmatched_dotdot_lhs,
-    bool remove_unmatched_dotdot_rhs) noexcept;
+    bool remove_unmatched_lhs,
+    bool remove_unmatched_rhs) noexcept;
```

Comment:
I still have no idea what these variables mean. Also, while the documentation for a public function goes in the declaration, the documentation for a private function goes in the definition. The reason, is that the user is only supposed to look at public declarations, while maintainers like us will only look at the definition (since that's where all the trouble is!)

This is why often you will see me leave out the formal parameter names in private function declarations, especially if they are obvious types given the context.

I need to know what these bools mean. How can a function that accepts constant views to a string buffer remove anything?

Reply:
I have no problem with any of that. I can rename anything or reinclude the comment explaining what this does. I've been doing this back and forth, I'm just not sure how to do it respecting the feng shui.

> I need to know what these bools mean. 

There's a discussion about this somewhere else. RFC determines we remove_dot_segments should remove unmatched ".." segments. But we need to turn this off for relative paths.